src/main/java/org/opensearch/sdk/handlers/ExtensionDependencyResponseHandler.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk.handlers;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.extensions.DiscoveryExtensionNode;
+import org.opensearch.extensions.ExtensionsManager;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportException;
+import org.opensearch.transport.TransportResponseHandler;
+
+/**
+ * This class handles the response from OpenSearch to a {@link ExtensionsRunner#sendExtensionDependencyRequest(TransportService)} call.
+ */
+public class ExtensionDependencyResponseHandler implements TransportResponseHandler<ExtensionDependencyResponse> {
+    private static final Logger logger = LogManager.getLogger(ExtensionDependencyResponseHandler.class);
+    private final CompleteableFuture<ExtensionDependencyResponse> inProgressFuture;
+    private List<DiscoveryExtensionNode> extensions;
+
+    public ExtensionDependencyResponseHandler() {
+        this.inProgressFuture = new CompletableFuture<>();
+        this.extensions = extensions.emptyList;
+    }
+
+    @Override
+    public void handleResponse(ExtensionDependencyResponse response) {
+        logger.info("received {}", response);
+
+        // Set cluster state from response
+        this.extensions = response.getExtensionDependencies();
+        inProgressFuture.complete(response);
+    }
+
+    @Override
+    public void handleException(TransportException exp) {
+        logger.info("ExtensionDependencyRequest failed", exp);
+        inProgressFuture.completeExceptionally(exp);
+    }
+
+    @Override
+    public String executor() {
+        return ThreadPool.Names.GENERIC;
+    }
+
+    @Override
+    public ExtensionDependencyResponse read(StreamInput in) throws IOException {
+        return new ExtensionDependencyResponse(in);
+    }
+
+    /**
+     * Invokes await on the ExtensionDependencyResponseHandler count down latch
+     * @throws Exception
+     *     if the response times out
+     */
+    public void awaitResponse() throws InterruptedException {
```

Comment:
Check which exceptions [`CompletableFuture.get()`](https://docs.oracle.com/en/java/javase/12/docs/api/java.base/java/util/concurrent/CompletableFuture.html#get()) throws.  In addition to the `InterruptedException` (rare in this case since the future will be completed before get is called) it will throw an `ExecutionException` so that should be added to the `throws`, as well as the javadoc `@throws` matching each one.

Reply:
I think `ExecutionException` still needs to be added, as well as `InterruptedException` to the javadocs